src/Controllers/SettingsController.php

```diff
@@ -21,11 +34,54 @@ class SettingsController extends BaseController
      * @param Response $response
      */
     public function __construct(
+        Authenticator $auth,
         Config $config,
+        LoggerInterface $log,
+        Redirector $redirector,
         Response $response
     ) {
-        $this->config = $config;
-        $this->response = $response;
+            $this->auth = $auth;
+            $this->config = $config;
+            $this->log = $log;
+            $this->redirect = $redirector;
+            $this->response = $response;
+    }
+
+    /**
+     * @return Response
+     */
+    public function password(): Response
+    {
+        return $this->response->withView(
+            'pages/settings/password.twig',
+            $this->getNotifications()
+        );
+    }
+
+    /**
+     * @return Response
+     */
+    public function savePassword(Request $request): Response
+    {
+        $user = $this->auth->user();
+
+        if (
```

Comment:
I would recommend using the internal validator here

```php
$minLength = config('min_password_length');
$data = $this->validate($request, [
    'password' => 'required',
    'new_password' => 'required|min:' . $minLength,
    'new_password2' => 'required'
]);
```

Eventually the is equal to can be validated, that has to be checked. I would definitely rename the field to password_confirmation ^^